gateway/modules/crud/dataloader.go

```diff
@@ -96,24 +96,24 @@ func newLoaderMap() *loaderMap {
 	return &loaderMap{m: map[string]*dataloader.Loader{}}
 }
 
-func (l *loaderMap) get(key string, graph *Module) *dataloader.Loader {
+func (l *loaderMap) get(key string, m *Module) *dataloader.Loader {
 	l.lock.Lock()
 	defer l.lock.Unlock()
 
 	if _, ok := l.m[key]; !ok {
-		l.m[key] = graph.createLoader()
+		l.m[key] = m.createLoader()
 	}
 
 	return l.m[key]
```

Comment:
Use two functions here: `getLoader` and `createLoader`. `getLoader` will return a loader and a boolean indicating whether its present or not. Make getLoader use a `RLock` to increase parallelism. `createLoader` will use a `Lock` internally since its doing a mutation